### DISCUSSION TITLE
Database Migrations

### DISCUSSION THREAD

**Discussion by DotNetDublin** [BlazingSaddlesSolutions/TimeKeeper#57] [General]
Creating a discussion to see if we can settle on an approach for managing database migrations (updates).

In terms of main stream options I think we have.
1. Code first and update database via entity framework migrations
2. Database via the SQL Server project and utilise the publish feature generate scripts

Missing anything?

This was first discussed as part of the [read me](https://github.com/BlazingSolutions/TimeKeeper/issues/42) issue

**Comment by AnkurSheel** [comment]
I dont know much about EF migrations but have we considered using [DbUp](https://dbup.readthedocs.io/en/latest/) instead of running migrations through SQL server?

**Reply by DotNetDublin** [reply]
I haven't come across DbUp before. Looks interesting.

**Reply by adamfoneil** [reply]
haven't heard of DbUp, either -- interesting!
